### PR TITLE
code restructure

### PR DESCRIPTION
**Pull request by maleck13**
Closes #

## Description of Changes
Largely moves some code around as per the proposal
slowly removing the reconcilers pkg

dnsrecord controller and reconciler is now in the dns package
add new v1.healthcheck type (not an API type yet but will likely become one in the future)
removed reconcilers pkg from the traffic pkg

the reconciler pkg is still there in the root for now but the hope is that ingress and route pkg can become a single traffic/controller.go and traffic/reconciler.go

## Remaining Work 

The following changes still need to be completed:

- [ ] List any outstanding work here

## Release and Testing

*

## Dependencies

- [ ] Perequisites that must be completed before merging this pull request

### CODE REVIEW

File: pkg/dns/controller.go [hunk 1]
```diff
@@ -136,25 +133,25 @@ func (c *Controller) process(ctx context.Context, key string) error {
...
-	var dnsProvider dns.Provider
-	provider, err := awsdns.NewProvider(awsdns.Config{})
-	if err != nil {
-		return nil, fmt.Errorf("failed to create AWS DNS manager: %v", err)
-	}
-	dnsProvider = provider
-
-	return dnsProvider, nil
-}
+// func (c *Controller) createDNSProvider(dnsProviderName string) (Provider, error) {
```
Comment (philbrookes) [inline pkg/dns/controller.go:136]: just delete this?

**Review by philbrookes — APPROVED** [review]

**Comment by philbrookes** [comment]
looks fine other than the failing e2e and lint tests

File: cmd/kcp-glbc/main.go [hunk 2]
```diff
@@ -39,21 +39,20 @@ import (
 	kcp "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
 	"github.com/kcp-dev/logicalcluster/v2"
 
+	"github.com/kuadrant/kcp-glbc/pkg/_internal/log"
```
Comment (mikenairn) [inline cmd/kcp-glbc/main.go:42]: Why "_internal" vs "internal"? Is that some convention I'm unaware of.

**Review by mikenairn — APPROVED** [review]
Question about the internal package name, but changes looks fine to me.

File: cmd/kcp-glbc/main.go (hunk 2 above)
Comment (maleck13) [reply cmd/kcp-glbc/main.go:42]: annoyingly, internal cannot be imported by main as it is in a different package. So I compromised and went with _internal to represent this stuff is the internal working but still allow it to be imported by the main

**Review by david-martin — COMMENTED** [review]
👍

**Pull request merged**